rest/rest-resources/src/main/java/org/apache/brooklyn/rest/util/MultiSessionAttributeAdapter.java

```diff
@@ -217,7 +219,14 @@ private void invalidateAllSession(HttpSession preferredSession, HttpSession loca
         private HttpSession findValidPreferredSession(HttpSession localSession, HttpServletRequest optionalRequest) {
             if (localSession instanceof Session) {
                 SessionHandler preferredHandler = getPreferredJettyHandler((Session)localSession, true, true);
-                HttpSession preferredSession = preferredHandler==null ? null : preferredHandler.getHttpSession(localSession.getId());
+
+                SessionIdManager idManager = preferredHandler.getSessionIdManager();
+                String id = idManager.getId(localSession.getId());
+                HttpSession preferredSession = preferredHandler.getSession(id);
+
+                if (preferredSession != null && !((Session)preferredSession).getExtendedId().equals(id))
```

Comment:
I think you're comparing to the wrong id here?  the localSession.getId returns the extended id and the idManager.getId returns the non-extended id